d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -636,6 +700,22 @@ public String toString()
       return "DiscoveryResponseData{" + "_resourceType=" + _resourceType + ", _resources=" + _resources + ", _nonce='"
           + _nonce + '\'' + '}';
     }
+
+    /**
+     * Invokes the given consumer for each resource in this response. If the {@link Resource} is not null, it is being
+     * created/modified and if it is null, it is being removed.
+     */
+    void foreach(BiConsumer<String, Resource> consumer)
```

Comment:
This was meant to mimic the `forEach` method that's available on `List` and `Map`. `consume` here does somehow imply that a subsequent call to `consumeResource` would yield nothing since they were consumed by the previous call. I changed it to `forEach`, no `foreach` to make it clearer, but I can change it to something else if you want